### ISSUE TITLE
Nav bar increased height

### ISSUE THREAD

**Issue by StefRe** [matplotlib/mpl-sphinx-theme#50]
On my 1440 x 900 pixels monitor the menu items in the new nav bar wrap into two lines so that the bar take much more space on the screen:

**OLD:**
![image](https://user-images.githubusercontent.com/19879328/190898042-1b93aa19-653a-4b4d-92f4-d4518a76e067.png)

**CURRENT:**
![image](https://user-images.githubusercontent.com/19879328/190898061-5f2be6c8-4761-470d-a69f-b58a3c4bf961.png)

Is there any chance to get this back into one line?

**Comment by timhoffm** [comment]
We have additional elements on the right, which at some point will start wrapping. However, that should not be happening already at 1440 pixels width.

You can remove the `#navbar-icon-links{ margin-left: ...}` CSS 
![image](https://user-images.githubusercontent.com/2836374/191009300-d74dcb8a-9c22-4e37-bef9-1ba74a332a3f.png)
and see if that already helps.

However, I have the feeling that something else is strange with your layout. Even with wrapping, the matplotlib logo should still be very much on the left.

**Comment by StefRe** [comment]
hm, I tested it on another computer with Windows 10 where I changed the resolution to 1440 x 900 and it indeed looks fine:

Firefox:
![grafik](https://user-images.githubusercontent.com/19879328/191027078-5f3eafd3-9299-477d-b0e4-e021789bcd20.png)

Edge:
![grafik](https://user-images.githubusercontent.com/19879328/191027712-1355f507-fd7d-4f70-bd93-984fd1ba3e37.png)


It obviously depends on the available fonts: on Linux the capital `E` of "Examples"  is 11 pixels high whereas on Window it's only 10 pixels high. Do you happend to have any advise on how I can fix it (I know this is not a matplotlib issue but maybe you could offhand quickly point me to the right direction)?

**Comment by jklymak** [comment]
On my 14" screen with the window 3/4 wide:  

<img width="1146" alt="Navbar" src="https://user-images.githubusercontent.com/1562854/191048601-8f942877-a140-4921-b7cb-d4b5ef6a4005.png">

**Comment by StefRe** [comment]
I can confirm that the reason for the issue are the different fonts - it's the only difference in the font properties of the nav bar text:

Windows (no wrapping):
![image](https://user-images.githubusercontent.com/19879328/191112385-bad223b6-634b-45aa-920c-18392eb9b574.png)


Linux (wrapping):
![image](https://user-images.githubusercontent.com/19879328/191112276-09460745-f76f-4f9f-a3df-a325761c0acc.png)

As a workaround I created a `userContent.css` file in the `<profile>/chrome` folder with the following contents
```
@-moz-document domain(matplotlib.org) {
    #navbar-main-elements .nav-link { font-size: 14px !important; }
}  
```
and set `toolkit.legacyUserProfileCustomizations.stylesheets` to `true`. 

It feels a bit hacky but it works fine.

**Comment by QuLogic** [comment]
It doesn't wrap for me on Linux; it appears to pick Liberation Sans instead. Maybe installing that would be a simpler fix.

**Comment by StefRe** [comment]
Is it possible to reduce the yellow extra space (see red arrow) between the logo and the nav bar links so that the links start a bit further to the left?

![image](https://user-images.githubusercontent.com/19879328/191116146-8fe57ef0-4638-425a-ad48-97b091432596.png)

**Comment by jklymak** [comment]
> Obviously not so many users working with 1440 pixel resolution on a 14" screen anymore.

I don't think this is quite correct - my screen has dpi-doubling (retina), so the screen resolution is quite high (3024 x 1964) and that 3/4 screen image was 2,268 pixels, but the pixel doubling makes css respond as if it is 1,134 pixels.  I think this is very common, and we should look at ways to reduce the width of our top bar, or design expecting this wrap.  

I could, of course, change the scaling on my display to make everything smaller, but what I use is the default.

**Comment by StefRe** [comment]
@QuLogic Thank you! This works too (installed [ttf-liberation](https://archlinux.org/packages/community/any/ttf-liberation/) and FF picks `Liberation Sans` that doesn't wrap at 15 px font size).

**Comment by timhoffm** [comment]
> we should look at ways to reduce the width of our top bar, or design expecting this wrap.

Agreed.

> Is it possible to reduce the yellow extra space

Possibly. However, this is introduced by pydata-sphinx-theme to align the nav element over the content, which makes sense to me on larger screens. So, this would have to adapt depending on the space.

As an alternative solution, I've propsed to move the version switcher next to the logo https://github.com/matplotlib/matplotlib/pull/23142, but that received mixed feedback.

The third option I could imagine is to turn the external links into a single <img width="32" alt="image" src="https://user-images.githubusercontent.com/2836374/191124507-c1e9ef0a-e05a-4562-94ab-eb08778a8121.png"> icon that opens a dropdown with all links. This would have to be responsive when the available size shrinks.

I don't think we want to limit or word-smith our regular nav elements just to become shorter.

**Comment by jklymak** [comment]
I don't mind the wrap on the icons so much, but the wrap on the nav items is hard to read and seems unnecessary (if you shrink much more, the theme switches to remove the nav altogether).  

Maybe we should just switch to the narrow screen theme at a wider size before the wrap happens.  However, the wrap is fontsize dependent - not sure how adaptive layout handles that.

**Comment by timhoffm** [comment]
> Maybe we should just switch to the narrow screen theme at a wider size before the wrap happens. However, the wrap is fontsize dependent - not sure how adaptive layout handles that.

I assume that font size only has a weak effect on wrapping, i.e. can shift it by a few pixels. If we configure the adaptive layout to switch a few pixels earlier than strictly needed by the fonts we have, that should be good enough.

**Comment by jklymak** [comment]
One top-nav issue I have is that Release Notes seems pretty niche.   I would think we are not unduly handicapping anyone by putting that under User docs.  I guess we could check the google stats and see how often this is hit directly.  But if I wanted our release notes I'd type that into a search engine before I'd navigate here.

**Comment by timhoffm** [comment]
I often browse to Release notes / Changelog pages to see when they released the last time, and what they changed. 🤷‍♂️

**Comment by QuLogic** [comment]
If I edit `--pst-font-family-base` (in browser) to ask for `DejaVu Sans`, then the headings wrap regardless of window width. It's just metrically larger than everything else:
![image](https://user-images.githubusercontent.com/302469/191145157-62daa198-18fd-4983-84e3-14c52c08edae.png)
Perhaps the theme should explicitly request some of these fonts as well?

**Comment by jklymak** [comment]
> I often browse to Release notes / Changelog pages to see when they released the last time, and what they changed. 🤷‍♂️ 

Sure, or consider rolling the plot styles, tutorials and examples in the user guide.  It feels like we have too many things at the top level nav.

**Comment by StefRe** [comment]
The icons on the right appear relatively big in comparison to the menu text in the nav bar - maybe it would already be enough to make the icons a bit smaller like in numpy or scipy:

Matplotlib:
![grafik](https://user-images.githubusercontent.com/19879328/191195129-5ba515df-dbe4-4406-91b7-108b58d0f5a2.png)


Numpy:
![grafik](https://user-images.githubusercontent.com/19879328/191195195-19257a2d-6076-4c05-8c03-b11ed77033af.png)


Scipy:
![grafik](https://user-images.githubusercontent.com/19879328/191195269-7db38675-9686-41a6-92ad-811ce3858031.png)

**Comment by jklymak** [comment]
> Could we try to put in for a small dev grant to do a real user study of the the website?

Maybe - but we can at least compare to what others do:

### pandas
<img width="1022" alt="Pandas" src="https://user-images.githubusercontent.com/1562854/191308269-48de9da6-c4b2-40e0-ad2c-b37e093f0e3a.png">

### numpy

<img width="1022" alt="numpy" src="https://user-images.githubusercontent.com/1562854/191308315-2b53793e-df5e-499c-b634-22ce8e4ac582.png">

So, fewer social, and fewer top-level nav.

**Comment by timhoffm** [comment]
> What about just shortening Release notes to releases?

`Releases` or `Changelog` are both fine by me.

That should fix the space problem for now.

**Comment by timhoffm** [comment]
Fixed by #51.
